### ISSUE TITLE
--no-window and -q aren't honored in beta12

### ISSUE THREAD

**Issue by thoraxe** [godotengine/godot#71380]
### Godot version

4.0-beta12

### System information

Linux (Fedora 35)

### Issue description

```
/opt/Godot/Godot_v4.0-beta12_mono_linux_x86_64/Godot_v4.0-beta12_mono_linux.x86_64 --no-window -q --build-solutions
```

The above command opens the window and fails to quit once the build is complete.

### Steps to reproduce

1. create a minimal c# project
2. attempt to build solutions from the CLI with `--no-window` and `-q`

### Minimal reproduction project

Can make a skeleton and try if needed.

**Comment by Calinou** [comment]
Did it work in 4.0.beta11? Please state it explicitly :slightly_smiling_face:

**Comment by akien-mga** [comment]
Check the `--help` output to know what flags are valid.

`--no-window` doesn't exist in 4.0. i
It's superseded by `--headless`.

`-q` is for "quiet" mode, not quit. You're looking for `--quit`.
